drivers/sensors/pmw33xx_common.c

```diff
@@ -0,0 +1,207 @@
+// Copyright 2022 Stefan Kerkmann (KarlK90)
+// Copyright 2021 Alabastard (@Alabastard-64)
+// Copyright 2020 Christopher Courtney, aka Drashna Jael're  (@drashna) <drashna@live.com>
+// Copyright 2019 Sunjun Kim
+// Copyright 2020 Ploopy Corporation
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "debug.h"
+#include "pmw33xx_common.h"
+#include "pointing_device_internal.h"
+#include "print.h"
+#include "string.h"
+#include "wait.h"
+#include "spi_master.h"
+#include "progmem.h"
+
+extern const uint8_t pmw33xx_firmware_data[PMW33XX_FIRMWARE_LENGTH] PROGMEM;
+extern const uint8_t pmw33xx_firmware_signature[3] PROGMEM;
+
+static const pin_t pmw33xx_cs_pins[]                                   = PMW33XX_CS_PINS;
+static bool        in_burst[(sizeof(pmw33xx_cs_pins) / sizeof(pin_t))] = {0};
+const size_t       pmw33xx_number_of_sensors                           = (sizeof(pmw33xx_cs_pins) / sizeof(pin_t));
+
+bool pmw33xx_spi_start(uint8_t sensor);
+void __attribute__((cold)) pmw33xx_upload_firmware(uint8_t sensor);
+bool __attribute__((cold)) pmw33xx_check_signature(uint8_t sensor);
+
+void pmw33xx_set_cpi_all_sensors(uint16_t cpi) {
+    for (uint8_t sensor = 0; sensor < pmw33xx_number_of_sensors; sensor++) {
+        pmw33xx_set_cpi(sensor, cpi);
+    }
+}
+
+bool pmw33xx_spi_start(uint8_t sensor) {
+    if (!spi_start(pmw33xx_cs_pins[sensor], false, 3, PMW33XX_SPI_DIVISOR)) {
```

Comment:
why did you drop using the defines like PMW3360_SPI_LSBFIRST, PMW3360_SPI_MODE here?

Reply:
We can be sure that the sensor are always *not* LSB first, so this can be a fixed value. I'm not 100% sure about the SPI mode but it should also always be mode 3.